app/helpers/foods_helper.rb

```diff
@@ -6,4 +6,49 @@ def item_avaiable food
   def load_categories
     @categories.map{|category| [category.name, category.id]}
   end
+
+  def subtotal_cart_value cart_items
+    subtotal_value = 0
+
+    cart_items.each do |item|
+      subtotal_value += item["price"] * item["food_quantity"]
+    end
+
+    subtotal_value
+  end
+
+  def increment_quantity cart_item
+    cart_item["food_avaiable_item"] -= params[:quantity].to_i
```

Comment:
logic chỗ này là khi user add sp vào giỏ hàng thì trừ Số lượng sp đi ấy hả e?

Reply:
logic là người dùng add thêm bao nhiêu sản phẩm thì số lượng avaiable_item lưu trữ trong sessions sẽ giảm đi để hiển thị số còn lại lên view ạ

Reply:
Sau khi cập nhật hiển thị trên view của user thì có xử lý gì tiếp theo ko e? 
Nếu user add sp vào giỏ hàng và KO có action thanh toán thì sp này sẽ ntn?

Reply:
Dạ chỉ để xử lý cho việc check validate số lượng user add vào giỏ hàng thôi ạ. Còn nếu người dùng không thanh toán thì sẽ không thực thi câu lệnh gì liên quan đến database cả nên số lượng sản phẩm đó trong database k có thay đổi ạ. Mục đích của em là em muốn check được rằng, nếu sản phẩm đó số lượng còn là 2, người dùng add 1 vào cart thì sẽ hiển thị trên view là còn 1 sản phẩm, nhưng chỉ lưu trữ tạm chứ không động gì đến dữ liệu trong database (vì chưa có action thanh toán) ạ. Tương tự nếu người dùng delete đi thì mình cũng sẽ cập nhật trong session tương tự ạ.